enrollment_predictions/models/auto_regressor_dec_tree.py

```diff
@@ -3,52 +3,123 @@
-        data['std_prev_{}'.format(win)] = course_groups['Enrolled'].transform(lambda x: x.shift().rolling(window=win).std())
-        data['ewm_{}'.format(win)] = course_groups['Enrolled'].transform(lambda x: x.shift().ewm(span=win).mean())
+        data.loc[:, 'mean_prev_{}'.format(win)] = course_groups['enrolled'].transform(lambda x: x.shift().rolling(window=win).mean())
+        data.loc[:, 'median_prev_{}'.format(win)] = course_groups['enrolled'].transform(lambda x: x.shift().rolling(window=win).median())
+        data.loc[:, 'min_prev_{}'.format(win)] = course_groups['enrolled'].transform(lambda x: x.shift().rolling(window=win).min())
+        data.loc[:, 'max_prev_{}'.format(win)] = course_groups['enrolled'].transform(lambda x: x.shift().rolling(window=win).max())
+        data.loc[:, 'std_prev_{}'.format(win)] = course_groups['enrolled'].transform(lambda x: x.shift().rolling(window=win).std())
+        data.loc[:, 'ewm_{}'.format(win)] = course_groups['enrolled'].transform(lambda x: x.shift().ewm(span=win).mean())
 
     return data
 
 
 def prepare_data(data):
     # Apply feature engineering to the train and validation sets separately
-    window_sizes = [2, 3, 6, 9]
-    
+    window_sizes = [1,2, 3, 6, 9]
+    #window_sizes = []
     data = feature_engineering(data, window_sizes)
-    data = remove_unnecessary_columns(data)
-    
-    exclude_columns = ['Year', 'Season', 'Enrolled'] 
+
+    exclude_columns = ['course', 'enrolled'] 
 
     train_features = data.columns[~data.columns.isin(exclude_columns)]
-    train_target = ['Enrolled', 'CourseOffering'] 
+    train_target = ['enrolled', 'offering'] 
+    
+    term_mapping = {'spring': 1, 'summer': 2, 'fall': 3}
+    data['term'] = data['term'].map(term_mapping)
+    
+    X_train = data[train_features].drop(columns=['offering'])
     
-    # Exclude 'CourseOffering' before imputation
-    X_train = data[train_features].drop(columns=['CourseOffering'])
+    X_train = X_train.dropna(axis=1, how='all')
+    X_train = X_train.loc[:, (X_train != "").any(axis=0)]
 
-    # Impute missing values
     imp = SimpleImputer()
     X_train_imputed = imp.fit_transform(X_train)
 
     return pd.DataFrame(X_train_imputed, columns=X_train.columns), data[train_target]
+
+
+def predict_year(model, predict_data, train_data, model_features):
+    new_data = pd.concat([train_data, predict_data])
+    new_data = new_data.sort_values('year')
     
-def predict_year(model, X_predict_year):
-    y_pred = model.predict(X_predict_year)
-    predictions_df = pd.DataFrame({'Predicted': y_pred})
-    return predictions_df
+    new_data_X, new_data_y = prepare_data(new_data)
+    prediction_year = predict_data['year'].iloc[0]
+    new_data_X = new_data_X[new_data_X['year'] == prediction_year]
+    new_data_X = new_data_X[model_features]
+
+    y_pred = model.predict(new_data_X)
+    
+    return pd.DataFrame({'Predicted': y_pred}, index=new_data_X.index)
```

Comment:
If there is something wrong with feature_engineering(), fix it and re-run the model to see if results improve. Thank you.